include/logging/log_core.h

```diff
@@ -578,7 +660,14 @@ void log_hexdump_sync(struct log_msg_ids src_level, const char *metadata,
  *
  * @note This function is intended to be used when porting other log systems.
  */
-void log_generic(struct log_msg_ids src_level, const char *fmt, va_list ap);
+void log_generic(struct log_msg_ids src_level, const char *fmt, va_list ap, bool enable_strdup);
```

Comment:
please add description of parameters to doxygen. Legacy parameters too which were missing.